From 53b424f64fb43a2f87229f30d28eb92b593f23c4 Mon Sep 17 00:00:00 2001
From: Nikos Mavrogiannopoulos <nmav@gnutls.org>
Date: Tue, 13 Aug 2013 17:47:24 +0300
Subject: [PATCH 250/291] cryptodev: Add

The attached patch contains the required modifications in order to get and set the cryptodev_verbosity variable using the following commands

root@OCSME:~# sysctl ioctl.cryptodev_verbosity
ioctl.cryptodev_verbosity = 0
root@OCSME:~# sysctl ioctl.cryptodev_verbosity=3
ioctl.cryptodev_verbosity = 3

Patch by Nikolaos Tsakalakis (nikolaos.tsakalakis@gmail.com).
---
 drivers/staging/crypto/cryptodev/ioctl.c | 28 ++++++++++++++++++++++++++++
 1 file changed, 28 insertions(+)

diff --git a/drivers/staging/crypto/cryptodev/ioctl.c b/drivers/staging/crypto/cryptodev/ioctl.c
index f26cf934e97f..f84b79c81ed6 100644
--- a/drivers/staging/crypto/cryptodev/ioctl.c
+++ b/drivers/staging/crypto/cryptodev/ioctl.c
@@ -45,6 +45,9 @@
 #include <linux/uaccess.h>
 #include <crypto/cryptodev.h>
 #include <linux/scatterlist.h>
+
+#include <linux/sysctl.h>
+
 #include "cryptodev_int.h"
 #include "zc.h"
 #include "version.h"
@@ -1101,6 +1104,26 @@ cryptodev_deregister(void)
 }
 
 /* ====== Module init/exit ====== */
+static struct ctl_table verbosity_ctl_dir[] = {
+        {
+                .procname       = "cryptodev_verbosity",
+                .data           = &cryptodev_verbosity,
+                .maxlen         = sizeof(int),
+                .mode           = 0644,
+                .proc_handler   = proc_dointvec,
+        },
+        {0, },
+};
+
+static struct ctl_table verbosity_ctl_root[] = {
+       {
+                .procname       = "ioctl",
+                .mode           = 0555,
+                .child          = verbosity_ctl_dir,
+       },
+       {0, },
+};
+static struct ctl_table_header *verbosity_sysctl_header;
 static int __init init_cryptodev(void)
 {
 	int rc;
@@ -1117,6 +1140,8 @@ static int __init init_cryptodev(void)
 		return rc;
 	}
 
+	verbosity_sysctl_header = register_sysctl_table(verbosity_ctl_root);
+
 	printk(KERN_INFO PFX "driver %s loaded.\n", VERSION);
 
 	return 0;
@@ -1127,6 +1152,9 @@ static void __exit exit_cryptodev(void)
 	flush_workqueue(cryptodev_wq);
 	destroy_workqueue(cryptodev_wq);
 
+	if (verbosity_sysctl_header)
+        unregister_sysctl_table(verbosity_sysctl_header);
+
 	cryptodev_deregister();
 	printk(KERN_INFO PFX "driver unloaded.\n");
 }
-- 
2.1.0

